sdk/pinocchio/src/entrypoint/lazy.rs

```diff
@@ -214,10 +206,10 @@ impl InstructionContext {
     /// before reading all accounts will result in undefined behavior.
     #[inline(always)]
     pub unsafe fn instruction_data_unchecked(&self) -> &[u8] {
-        let data_len = *(self.input.add(self.offset) as *const usize);
-        // shadowing the offset to avoid leaving it in an inconsistent state
-        let offset = self.offset + core::mem::size_of::<u64>();
-        core::slice::from_raw_parts(self.input.add(offset), data_len)
+        let data_len = *(self.input as *const usize);
+        // shadowing the input to avoid leaving it in an inconsistent position
+        let input = self.input.add(core::mem::size_of::<u64>());
+        core::slice::from_raw_parts(input, data_len)
```

Comment:
*minor*

When I was reading this code at first, it seemed to me that the function updates `self.input`, due to the `input` name being reused.
Considering this code is constructing a slice that covers the instruction data, perhaps it would be better not to use the `input` name again, and call it `data`, for example:

```suggestion
        let data = self.input.add(core::mem::size_of::<u64>());
        core::slice::from_raw_parts(data, data_len)
```
---
P.S. Probably out of scope for this PR, but why not import `size_of` as it seems to be used quite a lot in this module?

```rust
    use {
        /* ... */
        core::mem::size_of,
        /* ... */
    };

    /* ... */

        let data = self.input.add(size_of::<u64>());
        core::slice::from_raw_parts(data, data_len)
```

Reply:
Good call on both!